### ISSUE TITLE
app killed and calling from native log ios

### ISSUE THREAD

**Issue by andrewvmail** [WebsiteBeaver/CordovaCall#16]
not sure if its my just my build, if I kill the app and make a call from the native call log. i dont get any cordovacall event firing. it could be im using ionic wkwebview?

**Comment by andrewvmail** [comment]
i just wrote a workaround, i added a currentCallId attribute in CordovaCall class and just assign a value in the receiveCallFromRecents and then added a method called currentCall which just gives the value of that currentCallId expose it to JS. so after on document on ready i know who to call by calling that new method. is this too hacky to be a PR?

**Comment by dmarcs** [comment]
@andrewvmail You should be able to force quit and make a call from the call log. I'm glad you got it to work though. Let me play around with it tomorrow, and see if it gives me any problems. If it really doesn't work, you can definitely send a PR. Much appreciated.

**Comment by dmarcs** [comment]
@andrewvmail You're right this is a bug. If you make a call from recents after force quitting the app, the event handler doesn't get called. I'm going to look into it some more, and see if the way you fixed it is the best solution.

**Comment by dmarcs** [comment]
Fixed in CordovaCall 1.0.6. Great job catching this problem @andrewvmail. Your solution is very creative, and should work well for anyone who wants to implement it this way. I'd prefer not to add any new functions though, so if you use the event handler it will trigger whether or not you force quit the app.
```
cordova.plugins.CordovaCall.on('sendCall',function(info){
//info now contains the user id of the person you're trying to call
   setTimeout(function(){
       cordova.plugins.CordovaCall.connectCall();
    }, 500);
    alert(JSON.stringify(info));
});
```
I thought of 2 ways to solve this bug. Both aren't perfect unfortunately, but they work well enough IMO:

1. call
```
window.setTimeout(function(){
   exec(null, null, "CordovaCall", "jsLoaded", []);
}, 500);
```
at the end of [CordovaCall.js](https://github.com/WebsiteBeaver/CordovaCall/blob/master/www/CordovaCall.js). Create a global boolean variable in the Objective C code that keeps track of whether or not that JavaScript file finished loading. If it's false at the time of receiving a call from recents, assume that the app was closed. If it's true, assume that the app was already open. This solution is a variation of the solution that [Phonegap PushPlugin](https://github.com/phonegap/phonegap-plugin-push/blob/master/src/ios/PushPlugin.m#L326) uses to solve the same problem. The flaw with this is that the JavaScript might not finish loading within 500 milliseconds (but it probably will). When I tested it out, it worked 100% of the time. Also the end of the [CordovaCall.js](https://github.com/WebsiteBeaver/CordovaCall/blob/master/www/CordovaCall.js) file is not the best place to declare that the JavaScript finished loading. Really it would be best to add this at the end of `onDeviceReady`, but even then all of the even handlers might not be bound yet.

2. When `receiveCallFromRecents` is called, store the data in a global variable. When the user makes a call from Recents, check to see if the `sendCall` event is bound. If it is then continue normally as in CordovaCall 1.0.4 because this use case didn't give any issues. If the `sendCall` event isn't bound yet, then wait until it is. As soon as this happens through the call of `registerEvent`, take the data from the global variable and use it to trigger the event handler immediately. This solution isn't perfect because there's a problem if you try to bind the `sendCall` event after you make a call from recents (which is extremely unlikely because it makes no sense to do that).

I ended up going with solution 2 because I feel like no one will implement the use case that can break it. Solution 2 can be improved by clearing the global variable as soon as it's used once, but that assumes that the `sendCall` event only gets bound once (I don't see why anyone would bind it more than once). Solution 1 isn't bad either because most likely the event handlers will finish binding within 500 milliseconds.

As far as I know there really isn't a foolproof solution to this bug, but solution 2 will work for the majority of users, so I feel comfortable enough releasing it. If anyone has a better solution you're more than welcome to share it.
